conan/tools/cmake/toolchain/blocks.py

```diff
@@ -919,6 +920,65 @@ def _is_apple_cross_building(self):
         return os_host in ('iOS', 'watchOS', 'tvOS', 'visionOS') or (
                 os_host == 'Macos' and (arch_host != arch_build or os_build != os_host))
 
+    def _get_darwin_version(self, system_name, os_version):
```

Comment:
This could be simplified, you can check ranges with the Version class and use a dictionary like this:

```
        version_mapping = {
            "Macos": {
                "10.6": "10", "10.7": "11", "10.8": "12",
                ...
            }
        }
        return version_mapping.get(os_name, {}).get(os_version)
```